### ISSUE TITLE
[SSR error] environment.setHelperManager is not a function

### ISSUE THREAD

**Issue by lifeart** [josemarluedke/glimmer-apollo#45]
![image](https://user-images.githubusercontent.com/1360552/137622746-01659d8f-a3bd-4694-b49f-184381de2d51.png)

I got this if trying to render glimmerx app in ssr mode:

```js
import { renderToString } from '@glimmerx/ssr';

import App from './App';

export function render() {
    return renderToString(App, {
        rehydrate: true,
    });
}
```

I see environment resolution problems in SSR mode, especially, 

`import 'glimmer-apollo/environment-glimmer'` not working in nodejs (module not found), looks like it should be bundled to support nodejs execution or such

**Comment by lifeart** [comment]
reproduction app: https://github.com/lifeart/_glimmerx-apollo-ssr-issue

**Comment by josemarluedke** [comment]
@lifeart After a bit of trial and error I saw the Vite is not using conditional exports correctly when doing SSR.

If you change the import to `import 'glimmer-apollo/dist/cjs/environment-glimmer';`, you will note that for the most part, it will work. After the SSR page is served, then we get the error for the client not finding the module. There were a few issues in Vite around this issue, but they seem to be resolved/closed. 

I did find another issue when running in node using Vite, a PR for that is coming: `ReferenceError: self is not defined`.
